src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/SaveTempDataFilter.cs

```diff
@@ -29,16 +29,6 @@ public SaveTempDataFilter(ITempDataDictionaryFactory factory)
         /// <inheritdoc />
         public void OnResourceExecuting(ResourceExecutingContext context)
         {
```

Comment:
I see what you're saying here, but I'm not sure there's a solution. The issue is when you're using `IKeepTempDataResult` which mainly means it's a redirect. You don't write to the body and then redirect..... Let's discuss this and come to a decision about whether or not it's good enough